module/VuFind/src/VuFind/RecordDriver/SolrOverdrive.php

```
@@ -549,4 +562,39 @@ public function getRealTimeTitleHold()
         ];
         return $urlDetails;
     }
+
+    /**
+     * Return an array of associative URL arrays with one or more of the following
+     * keys:
+     *
+     * <li>
+     *   <ul>desc: URL description text to display (optional)</ul>
+     *   <ul>url: fully-formed URL (required if 'route' is absent)</ul>
+     *   <ul>route: VuFind route to build URL with (required if 'url' is absent)</ul>
+     *   <ul>routeParams: Parameters for route (optional)</ul>
+     *   <ul>queryString: Query params to append after building route (optional)</ul>
+     * </li>
+     *
+     * @return array
+     */
+    public function getURLs()
+    {
+        return $this->getIsMarc()
+            ? parent::getURLs()
+            : $this->getPermanentLink();
+    }
+
+    /**
+     * Get Permanent Link to the resource on your institution's OverDrive site
+     *
+     * @return string the permanent link to the resource
+     */
+    public function getPermanentLink()
+    {
+        $libraryURL = $this->config->libraryURL;
```

Where does this libraryURL setting come from? I don't see it in Overdrive.ini. Am I looking in the wrong place, or do we need to add an example? Also, do we need to handle the possibility that the setting is missing? Should we throw an exception in that case, or just skip providing a URL?

---

Strange. yes it was missing before and I am sure I've added it at least once. I guess it somehow didn't get committed. I think we do need the URL since it's library specific. I'll re-add it to config.